### ISSUE TITLE
[opencv4[cuda]] build failure

### ISSUE THREAD

**Issue by higurekon** [microsoft/vcpkg#16734]
**Host Environment**
- OS: Windows
- Compiler: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

**To Reproduce**
`./vcpkg install opencv4[cuda]:x64-windows`

**Failure logs** 
Snippet from console output:
```
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:144 (message):
    Command failed: C:/src/vcpkg/downloads/tools/cmake-3.19.2-windows/cmake-3.19.2-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v -j9
    Working Directory: C:/src/vcpkg/buildtrees/opencv4/x64-windows-dbg
    See logs for more information:
      C:\src\vcpkg\buildtrees\opencv4\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:105 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:45 (vcpkg_build_cmake)
  ports/opencv4/portfile.cmake:385 (vcpkg_install_cmake)
  scripts/ports.cmake:142 (include)
```
Corresponding error in install-x64-windows-dbg-out.log:
```
FAILED: bin/opencv_dnnd.dll lib/opencv_dnnd.lib 
cmd.exe /C "cd . && C:\src\vcpkg\downloads\tools\cmake-3.19.2-windows\cmake-3.19.2-win32-x86\bin\cmake.exe -E vs_link_dll --intdir=modules\dnn\CMakeFiles\opencv_dnn.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\COMMUN~1\VC\Tools\MSVC\1428~1.299\bin\Hostx64\x64\link.exe  @CMakeFiles\opencv_dnn.rsp  /out:bin\opencv_dnnd.dll /implib:lib\opencv_dnnd.lib /pdb:bin\opencv_dnnd.pdb /dll /version:4.5 /machine:x64  /nologo    /debug /INCREMENTAL  /NODEFAULTLIB:libc /DEBUG  && cd ."
LINK Pass 1: command "C:\PROGRA~2\MICROS~2\2019\COMMUN~1\VC\Tools\MSVC\1428~1.299\bin\Hostx64\x64\link.exe @CMakeFiles\opencv_dnn.rsp /out:bin\opencv_dnnd.dll /implib:lib\opencv_dnnd.lib /pdb:bin\opencv_dnnd.pdb /dll /version:4.5 /machine:x64 /nologo /debug /INCREMENTAL /NODEFAULTLIB:libc /DEBUG /MANIFEST /MANIFESTFILE:modules\dnn\CMakeFiles\opencv_dnn.dir/intermediate.manifest modules\dnn\CMakeFiles\opencv_dnn.dir/manifest.res" failed (exit code 1120) with the following output:
   Creating library lib\opencv_dnnd.lib and object lib\opencv_dnnd.exp
convolution_layer.cpp.obj : error LNK2019: unresolved external symbol cudnnGetConvolutionForwardAlgorithm referenced in function "public: __cdecl cv::dnn::cuda4dnn::csl::cudnn::ConvolutionAlgorithm<float>::ConvolutionAlgorithm<float>(class cv::dnn::cuda4dnn::csl::cudnn::Handle const &,class cv::dnn::cuda4dnn::csl::cudnn::ConvolutionDescriptor<float> const &,class cv::dnn::cuda4dnn::csl::cudnn::FilterDescriptor<float> const &,class cv::dnn::cuda4dnn::csl::cudnn::TensorDescriptor<float> const &,class cv::dnn::cuda4dnn::csl::cudnn::TensorDescriptor<float> const &)" (??0?$ConvolutionAlgorithm@M@cudnn@csl@cuda4dnn@dnn@cv@@QEAA@AEBVHandle@12345@AEBV?$ConvolutionDescriptor@M@12345@AEBV?$FilterDescriptor@M@12345@AEBV?$TensorDescriptor@M@12345@3@Z)
convolution_layer.cpp.obj : error LNK2019: unresolved external symbol cudnnGetConvolutionBackwardDataAlgorithm referenced in function "public: __cdecl cv::dnn::cuda4dnn::csl::cudnn::TransposeConvolutionAlgorithm<float>::TransposeConvolutionAlgorithm<float>(class cv::dnn::cuda4dnn::csl::cudnn::Handle const &,class cv::dnn::cuda4dnn::csl::cudnn::ConvolutionDescriptor<float> const &,class cv::dnn::cuda4dnn::csl::cudnn::FilterDescriptor<float> const &,class cv::dnn::cuda4dnn::csl::cudnn::TensorDescriptor<float> const &,class cv::dnn::cuda4dnn::csl::cudnn::TensorDescriptor<float> const &)" (??0?$TransposeConvolutionAlgorithm@M@cudnn@csl@cuda4dnn@dnn@cv@@QEAA@AEBVHandle@12345@AEBV?$ConvolutionDescriptor@M@12345@AEBV?$FilterDescriptor@M@12345@AEBV?$TensorDescriptor@M@12345@3@Z)
bin\opencv_dnnd.dll : fatal error LNK1120: 2 unresolved externals
```
**Additional context**
I cloned vcpkg today, so I should have the latest version. I had CUDA toolkit v11.2 installed today as well, and made sure to reboot before trying to install opencv4[cuda] with vcpkg. My GPU is a Geforce GTX Gigabyte 1080 Ti card.

**Comment by JackBoosY** [comment]
cc @cenit

**Comment by cenit** [comment]
cudnn installation mismatch?
#16413 might be important to fix this issue

**Comment by higurekon** [comment]
Ah yes.

I have since found the fix and yes, my issue is a duplicate of #16413 and #16636
